src/Http/Routing/src/Builder/EndpointRouteBuilderExtensions.cs

```diff
@@ -495,8 +498,28 @@ public static RouteHandlerBuilder MapFallback(this IEndpointRouteBuilder endpoin
         // explicit about the MethodInfo representing the "handler" and not the RequestDelegate?
 
         // Add MethodInfo as metadata to assist with OpenAPI generation for the endpoint.
+        // This is added directly to the builder to ensure it's first in the metadata collection.
         builder.Metadata.Add(handler.Method);
 
+        // Defer adding attributes on the handler to the call to RDF as those automatically generated by the
+        // RDF have a higher specificity.
+        var attributes = handler.Method.GetCustomAttributes();
+
+        // This can be null if the delegate is a dynamic method or compiled from an expression tree
+        if (attributes is not null)
+        {
+            foreach (var attribute in attributes)
+            {
+                metadataToAdd.Add(attribute);
```

Comment:
What do you think about adding the `MethodInfo`'s attributes at the start of `RequestDelegateFactory.Create()` instead of here? I think it makes sense to have RDF do it now that it's adding a bunch of other metadata. That wasn't the case originally.

Reply:
Yeah that makes sense. This would be a change for direct callers of `Create` however but is likely OK in the scheme of things.

Reply:
Just realized I misread what you wrote as "adding the `MethodInfo` as metadata" and made a change based on that. So, what do we think about moving the addition of both the `MethodInfo` and the `MethodInfo`'s attributes as metadata into `RequestDelegateFactory.Create()`? We'd want to ensure the attributes are added last, after anything inferred, so that it's still possible to override any inferred metadata via an attribute (as the last metadata item of a given type is the most specific).